### ISSUE TITLE
Interpolations mess with source map

### ISSUE THREAD

**Issue by mxstbr** [styled-components/stylelint-processor-styled-components#1]
This will (incorrectly) report the indentation error to be on line 2 instead of 3.

```JS
const Button = styled.button`
  color: ${constants.color};
background: red;
`;
```

This is because we remove lines with interpolations, but don't account for them in the source maps fixing. This might need significant restructuring of the internals to fix. 😕

**Comment by davidtheclark** [comment]
> we remove lines with interpolations

Could you elaborate on what you mean by this? I don't think I understand.

One possibility to keep in mind, in case it's relevant, is that you could "blur" interpolated values — replace the characters with some innocuous character — so they are are unlikely to mess up stylelint but will still occupy the same number of characters. We use this hacky trick in stylelint sometimes to deal with SCSS interpolation. E.g. replace `color: ${constants.color};` with `color: xxxxxxxxxxxxxxxxxx;`. Does that seem relevant?

**Comment by mxstbr** [comment]
> you could "blur" interpolated values 

The problem is that you can interpolation _anything_, from a `css` fragment to a simple property:

```JS
const color = 'red'
const background = 'background: blue;'
const Button = styled.button`
  color: ${color};
  ${background}
  ${`
    color: blue;
  `}
`;
```

So blurring is really hard, because you essentially cannot know what to put into those interpolations.

----

Right now I literally just cut out the lines with interpolations before sending the string to stylelint, i.e. this:

```JS
const Button = styled.button`
  border: none;
  color: ${color};
  background: red;
`;
```

becomes this CSS for stylelint:

```CSS
.selector {
  border: none;
  background: red;
}
```

(the interpolation line is just removed)

The problem is that I do this cutting out when I get the contents of the string from babel, but I only do the actual source mapping (i.e. mapping the `border: none;` line in the CSS to the correct one in the JS) much later. 😕

**Comment by davidtheclark** [comment]
> The problem is that you can interpolation anything, from a css fragment to a simple property:

How about not telling stylelint what's in the interpolation? What you would "blur" would be the interpolation itself `${..}` — the `$`, `{}`, and everything between the brackets.

In fact, you might not even need to "blur", because those interpolations look kind of like SCSS variables, and *stylelint rules are built to ignore SCSS variables*. So in your example above, try just leaving `${color}` as-is, see what happens.

**Comment by mxstbr** [comment]
Hmm, I tried that in the [`leave-interpolations`](https://github.com/styled-components/stylelint-processor-styled-components/tree/leave-interpolations) branch, but stylelint throws this error:

```JSON
[{
  line: 3,
  column: 12,
  rule: 'CssSyntaxError',
  severity: 'error',
  text: 'Unknown word (CssSyntaxError)'
}]
```

When it sees this CSS:

```CSS
.selector {
  background: red;
  color: ${color};
  display: block;
}
```

Any ideas what I'm doing wrong?

**Comment by davidtheclark** [comment]
@mxstbr Need to run stylelint with SCSS syntax: `stylelint --syntax scss`. Give that a shot.

**Comment by mxstbr** [comment]
Hah, that works for simple cases (like `${color}`) if I remove the `{}`! Let me try some more complex examples, but I like where this is going 👌

**Comment by mxstbr** [comment]
It works for complex cases too (those just return `undefined`, which is totally fine) – I think we have a winner here, this works perfectly so far! I'll merge and ship this in a new version!

**Comment by mxstbr** [comment]
Awesome, got it working! Thanks!

**Comment by davidtheclark** [comment]
@mxstbr: Fantastic. Since stylelint tries to deal with a variety of non-standard syntaxes, there are always bound to be little bugs that pop up here and there when people do something especially complex (e.g. SCSS equations with variables inside interpolation etc etc). So if these bugs start popping up for this tool, please encourage users to help stylelint figure out what little tweak needs to be made to ignore what needs to be ignored. 

However, this tool might be uniquely protected against these bugs because everything non-standard is going to be wrapped in `${..}`. We'll see! Great work with this!

**Comment by mxstbr** [comment]
Thank you so much for your processor and the help, would not have gotten this done without you!
